Update base image

jburel opened the pull request:
Attempting to get the BioFormats-image job green in new jenkins-CI
Currently failure due to 
```
Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project ome-common: There are test failures.
```
The change also matches the arguments in BF-build job

jburel commented:
Looking at the Image
There is a difference between the version of merge-ci and updated jenkins-ci
Recent update see https://github.com/docker-library/openjdk

sbesson commented:
Looking quickly at it locally, it looks like we are hitting https://stackoverflow.com/q/23260057. Amongst possible suggested solutions, one of them might be to revert https://github.com/ome/bio-formats-build/pull/58 (i.e. it looks like we had already had this issue in the past)

jburel commented:
the commit 441e5ea did not break the build on the updated infra

jburel commented:
When running locally I have errors when building the javadoc
``100 errors
#14 580.3 100 warnings
#14 580.3 [ERROR] MavenReportException: Error while generating Javadoc: 
#14 580.3 Exit code: 1 - /bio-formats-build/ome-poi/src/main/java/loci/poi/hpsf/ClassID.java:218: error: unexpected end tag: </p>
#14 580.3      * <code>false</code>.</p>
#14 580.3                           ^
#14 580.3 /bio-formats-build/ome-poi/src/main/java/loci/poi/hpsf/ClassID.java:51: error: malformed HTML
#14 580.3  * @author Rainer Klute <a
#14 580.3                         ^
#14 580.3 /bio-formats-build/ome-poi/src/main/java/loci/poi/hpsf/ClassID.java:52: error: bad use of '>'
#14 580.3  * href="mailto:klute@rainer-klute.de">&lt;klute@rainer-klute.de&gt;</a>
#14 580.3                                       ^
#14 580.3 /bio-formats-build/ome-poi/src/main/java/loci/poi/hpsf/ClassID.java:52: error: unexpected end tag: </a>
#14 580.3  * href="mailto:klute@rainer-klute.de">&lt;klute@rainer-klute.de&gt;</a>
#14 580.3                                                                     ^
#14 580.3 /bio-formats-build/ome-poi/src/main/java/loci/poi/hpsf/Constants.java:46: error: malformed HTML
#14 580.3  * @author Rainer Klute <a
#14 580.3                         ^
#14 580.3 /bio-formats-build/ome-poi/src/main/java/loci/poi/hpsf/Constants.java:47: error: bad use of '>'
#14 580.3  * href="mailto:klute@rainer-klute.de">&lt;klute@rainer-klute.de&gt;</a>
#14 580.3        
```

jburel commented:
@dgault have you seen this error before?

sbesson commented:
@jburel should the title of this PR be updated as it is allowing the tests to be run on new JDK versions rather than skipping them

dgault commented:
I've never seen any of those before, Im assuming there must be extra or stricter checking going on here. When I get some free time I will try to reproduce locally

jburel commented:
It looks like the image is older in the new devspace
```
docker inspect openjdk:8 | grep Created
        "Created": "2019-03-05T04:04:10.522942166Z",
```
and merge-ci
```
docker inspect openjdk:8 | grep Created
        "Created": "2021-07-22T13:37:31.954806499Z",

```
That probably explains what the divergence we noticed

sbesson commented:
So is the next step to pull a recent Docker image, exclude this and see if the problem is fixed?

jburel commented:
I have 
 * modified the Dockerfile so a recent image is pulled
 * removed flag re-introduced with [441e5ea](https://github.com/ome/bio-formats-build/pull/241/commits/441e5eabc2323ee1bb5981d28820ef325cad343a)

jburel commented on Dockerfile:
```diff
@@ -20,6 +20,7 @@ RUN git submodule update --init
 RUN python3 -m venv /bio-formats-build/venv
 ENV PATH="/bio-formats-build/venv/bin:$PATH"
 RUN pip install -r ome-model/requirements.txt
+RUN sphinx-build --version
```
This was introduced during local testing and I forgot to remove it

sbesson commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-ARG BUILD_IMAGE=openjdk:8
+ARG BUILD_IMAGE=openjdk:8u322-jdk
```
I know `docker` is one of the ecosystems supported by dependabot (https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates) but I have no experience with it and I don't know if that would apply to this use case

jburel commented:
@dgault what do you think about pinning the version?

dgault commented:
Fixing to a specific version seems like a sensible approach to me, should keep things more consistent. Looks good from my side.
